src/EnergyPlus/UnitarySystem.cc

```diff
@@ -0,0 +1,4202 @@
+                if (fields.find("single_mode_operation") != fields.end()) { // not required field
+                    loc_singleModeOp = UtilityRoutines::MakeUPPERCase(fields.at("single_mode_operation"));
+                }
+
+                Real64 loc_noLoadAirFlowRateRatio(1.0);
+                if (fields.find("no_load_supply_air_flow_rate_ratio") != fields.end()) { // not required field
+                    loc_noLoadAirFlowRateRatio = fields.at("no_load_supply_air_flow_rate_ratio");
+                }
+
+                // set JSON data to design spec object
+                thisDesignSpec.name = thisObjectName;
+                thisDesignSpec.designSpecMSHPType_Num = 1; // add global int value for factory
+                thisDesignSpec.numOfSpeedHeating = loc_numOfSpeedHeating;
+                thisDesignSpec.numOfSpeedCooling = loc_numOfSpeedHeating;
+
+                if (UtilityRoutines::SameString(loc_singleModeOp, "Yes")) {
+                    thisDesignSpec.singleModeFlag = true;
+                } else if (UtilityRoutines::SameString(loc_singleModeOp, "No")) {
+                    thisDesignSpec.singleModeFlag = false;
+                } else {
+                }
+
+                thisDesignSpec.MSHeatingSpeedRatio.resize(maxSpeeds);
+                thisDesignSpec.MSCoolingSpeedRatio.resize(maxSpeeds);
+                for (int speedNum = 0; speedNum < maxSpeeds; ++speedNum) {
+                    thisDesignSpec.MSHeatingSpeedRatio[speedNum] = 1.0;
+                    thisDesignSpec.MSCoolingSpeedRatio[speedNum] = 1.0;
+                    std::string tempString("heating_speed_" + std::to_string(speedNum + 1) + "_supply_air_flow_ratio");
+
+                    // these do not work since "heating_speed_x_supply_air_flow_ratio" fields are extensible and in "flow_ratios"
+                    // if (fields.find("heating_speed_" + std::to_string(speedNum+1) + "_supply_air_flow_ratio") != fields.end()) { // not required
+                    // field
+                    //    thisDesignSpec.MSHeatingSpeedRatio[speedNum] =
+                    //        fields.at("heating_speed_" + std::to_string(speedNum+1) + "_supply_air_flow_ratio");
+                    //}
+                    // if (fields.find("cooling_speed_" + std::to_string(speedNum+1) + "_supply_air_flow_ratio") != fields.end()) { // not required
+                    // field
+                    //    thisDesignSpec.MSCoolingSpeedRatio[speedNum] =
+                    //        fields.at("cooling_speed_" + std::to_string(speedNum+1) + "_supply_air_flow_ratio");
+                    //}
+                    // this find does work and finds "flow_ratios"
+                    if (fields.find("flow_ratios") != fields.end()) {
+                        // but how to get the data for heating and cooling speeds ratios?
+                        // thisDesignSpec.MSHeatingSpeedRatio[speedNum] = fields.at.array[speedNum]("heating_speed_supply_air_flow_ratio");
```

Comment:
@Myoldmopar how do I read in the cooling/heating speed data from these extensible fields? These data show up under "flow_ratios" in the field reference. The strings e.g. "Heating Speed 1 Supply Air Flow Ratio" are not there but there is an array of e.g. "Heating Speed Supply Air Flow Ratio" but I don't know how to access the data.

Reply:
@mbadams5 can you school me on how to JSON read the Heating/Cooling Speed x Air Flow Ratio data in the UnitarySystemPerformance:Multispeed object? This branch compiles so you can test right here if you like. I've tried and am just spinning my wheels.

Reply:
@mbadams5 this is in UnitarySystem.cc at line 228.

Reply:
@mbadams5 never mind @Myoldmopar found a method.